checker/worker/scheduler.go

```diff
@@ -0,0 +1,117 @@
+package worker
+
+import (
+	"time"
+
+	"github.com/moira-alert/moira"
+	"github.com/moira-alert/moira/checker"
+	"github.com/moira-alert/moira/metrics"
+	w "github.com/moira-alert/moira/worker"
+)
+
+const checkerLockTTL = time.Second * 15
+
+type scheduler struct {
+	manager           *WorkerManager
+	clusterKey        moira.ClusterKey
+	sourceCheckConfig checker.SourceCheckConfig
+	name              string
+	lockName          string
+	validateSource    func() error
+	metrics           *metrics.CheckMetrics
+}
+
+func newScheduler(manager *WorkerManager, clusterKey moira.ClusterKey, validateSource func() error) (*scheduler, error) {
+	metrics, err := manager.Metrics.GetCheckMetricsBySource(clusterKey)
+	if err != nil {
+		return nil, err
+	}
+
+	name := clusterKey.TriggerSource.String() + ":" + clusterKey.ClusterId.String()
```

Comment:
May be use ClusterKey.String() or some difference is fatal ?

Reply:
Done